sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -90,9 +98,9 @@ class RapidsHostMemoryStore(
         other.id,
         other.size,
         other.meta,
-        other.getSpillPriority,
+        other.getSpillPriority + allocationMode.spillPriorityOffset,
```

Comment:
This should probably clamp against the extremes to avoid overflow in case someone decides to create a device memory buffer at the lowest or highest possible priority, otherwise adding this adjustment have the opposite of the desired effect for those cases.

Reply:
Done.